server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymRuleAction.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.synonyms;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.ValidateActions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.synonyms.SynonymRule;
+import org.elasticsearch.synonyms.SynonymsManagementAPIService;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class PutSynonymRuleAction extends ActionType<SynonymUpdateResponse> {
+
+    public static final PutSynonymRuleAction INSTANCE = new PutSynonymRuleAction();
+    public static final String NAME = "cluster:admin/synonym_rules/put";
+
+    public PutSynonymRuleAction() {
+        super(NAME, SynonymUpdateResponse::new);
+    }
+
+    public static class Request extends ActionRequest {
+        private final String synonymsSetId;
+
+        private final SynonymRule synonymRule;
+
+        public static final ParseField SYNONYMS_FIELD = new ParseField(SynonymsManagementAPIService.SYNONYMS_FIELD);
+        private static final ConstructingObjectParser<SynonymRule, String> PARSER = new ConstructingObjectParser<>(
+            "synonyms",
```

Comment:
Nitpick: should this be singular? 

```suggestion
            "synonym",
```

Reply:
This way it has the same name as the field it reads, I think it's ok to leave it as is

Reply:
Here we are dealing with multiple synonyms, right? So "synonyms" is more appropriate.